boards/sltb001a/include/periph_conf.h

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (C) 2015-2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_sltb001a
+ * @{
+ *
+ * @file
+ * @brief       Configuration of CPU peripherals for the SLTB001A starter kit
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ */
+
+#ifndef PERIPH_CONF_H
+#define PERIPH_CONF_H
+
+#include "cpu.h"
+
+#include "periph_cpu.h"
+
+#include "em_cmu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    Clock configuration
+ * @{
+ */
+#ifndef CLOCK_HF
+#define CLOCK_HF            cmuSelect_HFXO
+#endif
+#ifndef CLOCK_CORE_DIV
+#define CLOCK_CORE_DIV      cmuClkDiv_1
+#endif
+#ifndef CLOCK_LFA
+#define CLOCK_LFA           cmuSelect_LFXO
+#endif
+#ifndef CLOCK_LFB
+#define CLOCK_LFB           cmuSelect_LFXO
+#endif
+#ifndef CLOCK_LFE
+#define CLOCK_LFE           cmuSelect_LFXO
+#endif
+/** @} */
+
+/**
+ * @name    ADC configuration
+ * @{
+ */
+static const adc_conf_t adc_config[] = {
+    {
+        ADC0,                               /* device */
+        cmuClock_ADC0,                      /* CMU register */
+    }
+};
+
+static const adc_chan_conf_t adc_channel_config[] = {
+    {
+        0,                                  /* device index */
+        adcPosSelTEMP,                      /* channel to use */
+        adcRef1V25,                         /* channel reference */
+        adcAcqTime8                         /* acquisition time */
+    },
+    {
+        0,                                  /* device index */
+        adcPosSelAVDD,                      /* channel to use */
+        adcRef5V,                           /* channel reference */
+        adcAcqTime8                         /* acquisition time */
+    }
+};
+
+#define ADC_NUMOF           (2U)
+#define ADC_0_EN            (1)
+#define ADC_1_EN            (1)
```

Comment:
I think it would be cleaner to use something like `#define ADC_DEV_NUMOF` instead of these enable defines.

Reply:
Where one device has multiple 'children', I defined `*_DEV_NUMOF` and `*_NUMOF`. I need the `*_DEV_NUMOF` for the assertions and locks.